services/forms/repo_form.go

```diff
@@ -162,6 +162,7 @@ type RepoSettingForm struct {
 
 	// Admin settings
 	EnableHealthCheck bool
+	Index             string
```

Comment:
The word `Index` is so short (and a little confusing), maybe `ReindexRepoIndexerType` is better, or at least `ReindexRepo`.

Otherwise LGTM

Reply:
I don't think we should allow repo admin to reindex the repository. It should be a responsibility of system admin.

Reply:
If I understand correctly, this PR is checking the user for admin permissions

```
	case "admin_index":
		if !ctx.User.IsAdmin {
			ctx.Error(http.StatusForbidden)
			return
		}
```

Reply:
But repository admin could see that button in the repository setting page

```go
if ctx.Repo.IsAdmin() {
```

Reply:
@wxiaoguang renamed RequestReindexType

@lunny now checks ctx.User.IsAdmin instead.

Reply:
But it's still wired a button on repository setting but only site administrator could click?

Reply:
It's not the only button that's there - the fsck option is there too.

Reply:
So maybe both them should be moved into admin repositories. I will not prevent this merge. But maybe we could change that in other PRs.